@xen-orchestra/rest-api/src/groups/group.controller.mts

```diff
@@ -49,4 +70,25 @@ export class GroupController extends XoController<XoGroup> {
   getGroup(@Path() id: string): Promise<Unbrand<XoGroup>> {
     return this.getObject(id as XoGroup['id'])
   }
+
+  /**
+   * @example body {
+   * "name": "new group",
+   * "provider": "64a7a0b4-e728-47e2-a082-93a218890a81",
+   * "providerGroupId": "722d17b9-699b-49d2-8193-be1ac573d3de"
+   * }
+   */
+  @Example(groupId)
+  @Post('')
+  @Middlewares(json())
+  @SuccessResponse(createdResp.status, createdResp.description)
+  @Response(resourceAlreadyExists.status, resourceAlreadyExists.description)
```

Comment:
Your endpoint never returns a `409` response.
It returns a `500` response with error message: `the group <group-name> already exists`.
Fix that by really throwing a `objectAlreadyExists` error in `xoApp.createGroup`

Reply:
Much better, thanks!
BTW, don't forget to document that your endpoint can return a `409` response